pkg/network/usm/tests/tracer_usm_kafka_classification_linux_test.go

```diff
@@ -271,13 +252,61 @@ func testKafkaProtocolClassification(t *testing.T, tr *tracer.Tracer, clientHost
 	}
 
 	for produceVersion := kafka.MinSupportedProduceRequestApiVersion; produceVersion <= kafka.MaxSupportedProduceRequestApiVersion; produceVersion++ {
-		currentTest := getProduceVersionTest(int16(produceVersion), targetAddress, serverAddress, getTopicName(), defaultDialer)
+		expectedProtocol := protocols.Kafka
+		if produceVersion < kafka.MinSupportedProduceRequestApiVersion || produceVersion > kafka.MaxSupportedProduceRequestApiVersion {
+			expectedProtocol = protocols.Unknown
+		}
+
+		version := kversion.V4_0_0()
+		targetPort := kafkaPort
+
+		// on older versions of kafka, test against old kafka server
+		if produceVersion < 8 {
+			version = kversion.V3_8_0()
+			targetPort = kafka.KafkaOldPort
+		}
```

Comment:
why is that?
Did kafka v4 broke support for old versions?

Reply:
Yep :(